components/zcash_address/src/kind/unified.rs

```diff
@@ -75,19 +76,25 @@ impl PartialOrd for Typecode {
     }
 }
 
-impl From<u8> for Typecode {
-    fn from(typecode: u8) -> Self {
-        match typecode {
-            0x00 => Typecode::P2pkh,
-            0x01 => Typecode::P2sh,
-            0x02 => Typecode::Sapling,
-            0x03 => Typecode::Orchard,
-            _ => Typecode::Unknown(typecode),
+impl TryFrom<u32> for Typecode {
+    type Error = ParseError;
+
+    fn try_from(typecode: u32) -> Result<Self, Self::Error> {
+        if typecode < 0x02000000 {
```

Comment:
This is the wrong bound:
https://github.com/zcash/librustzcash/blob/a3df9dd47d6c3fa89b08483cb8276515464de321/components/zcash_encoding/src/lib.rs#L57-L61